common/exception/GraknClientException.java

```diff
@@ -33,15 +34,17 @@ public GraknClientException(String error) {
         this.errorMessage = null;
     }
 
-    public GraknClientException(ErrorMessage error) {
-        super(error.toString());
+    public GraknClientException(ErrorMessage error, Object... parameters) {
+        super(error.message(parameters));
         assert !getMessage().contains("%s");
         this.errorMessage = error;
     }
 
     public static GraknClientException of(StatusRuntimeException statusRuntimeException) {
         if (statusRuntimeException.getStatus().getCode() == Status.Code.UNAVAILABLE) {
             return new GraknClientException(ErrorMessage.Client.UNABLE_TO_CONNECT);
+        } else if (statusRuntimeException.getStatus().getCode() == Status.Code.INTERNAL && statusRuntimeException.getStatus().getDescription().contains("[RFT01]")) {
```

Comment:
In this particular if block I want to check if the server throws an exception because there's no leader yet.

The server will throw an exception of code "[RFT01]", and the only way to propagate that information about the exception is by embedding it in the message, hence the need for `getDescription().contains(...)`.

Is there a better way to propagate exceptions than this?

Do you have better ideas?

Reply:
The best alternatives that we have _right now_ would be to either:

- change the protocol and server: don't throw an exception if there's no leader, but rather encode the error in protobuf; or
- add a new method to protocol and server to check if the leader exists - but this is both inefficient and non-atomic.

So in conclusion parsing the error message is the least bad option right now.

There are plans, in the future, to implement a dedicated `Error` message in protocol, which we will be able to interpret without the need for hacks: see https://github.com/graknlabs/client-java/issues/180 .

Reply:
I've moved the hacky code into its own method and added a TODO: https://github.com/graknlabs/client-java/pull/243/commits/268c884c5fae149d70c0136e643ba117e88c39f9